units/src/amount.rs

```diff
@@ -191,42 +191,39 @@ impl From<ParseDenominationError> for ParseAmountError {
     fn from(e: ParseDenominationError) -> Self { Self::InvalidDenomination(e) }
 }
 
+internals::parse_error_type! {
+    "An error during denomination parsing.",
+    pub ParseDenominationError(InvalidDenominationError, "a bitcoin denomination")
+}
+
 /// An error during amount parsing.
 #[derive(Debug, Clone, PartialEq, Eq)]
 #[non_exhaustive]
-pub enum ParseDenominationError {
+pub enum InvalidDenominationError {
     /// The denomination was unknown.
-    Unknown(String),
```

Comment:
This should've been changed to `internals::error::InputString` and then everything else should work, possibly with small changes. However I don't want to expose `InputString` as public so It'd be best to first wrap the inner errors in structs.

Reply:
Ah damn, and I thought I had it. Used as suggeseted.